src/mixins/data-iterable.js

```diff
@@ -240,11 +240,18 @@ export default {
         this.defaultPagination = updatedPagination
       }
     },
+    getItemKey (item) {
```

Comment:
I'll repeat my comment in case it was missed: `getItemKey` could be a computed property
`return typeof key === 'function' ? key : item => this.getObjectValueByPath(item, key)`

Reply:
I don't see any reason to be a computed property, `getItemKey` must always be a function so a method makes more sense. If it is a computed property there is an unnecessary proxy function to `getObjectValueByPath`. While debugging (or just reading code), if you find something like `this.something()` you'll think that it is a method not a prop. Also there is a semantic problem, if I make `getItemKey` a computed prop then I should rename it to something like `getItemKeyFunc` (to know that this prop will always give me a function).

So, even if it can be a computed prop, my opinion is not to make it one.

Reply:
> If it is a computed property there is an unnecessary proxy function to getObjectValueByPath

Now the `getItemKey` function itself is a proxy to `getObjectValueByPath` so one proxy will be just replaced with another one, but it will remove the proxy to the `key` function. And for both cases it will remove some code that is run every time the `getItemKey` is called like checking whether `key` is a function or not

>  I should rename it to something like getItemKeyFunc

I don't think it's a bigger problem than having a `key` prop which could be a function but you can't guess it just by looking at the name. And `get*` name clearly shows that this is a function.

Not sure if this solution would be debugging-friendly though

Reply:
I think that in most cases `itemKey` will be a string, and in some edge cases you'll need a function for custom behaviour. But you still need to allow a function to prop because a developer works with 3rd party APIs or have custom data structures, otherwise vuetify is good just for demos and examples, not for production apps. 
Thats why `itemValue` and `itemText` from `VSelect` also support functions.

Here is an example for `VSelect.itemValue`

```js
items = [
   {
       type: 'car',
       carTitle: '1st car',
       carId: 'car-1'
   },
   {
      type: 'bike',
      bikeTitle: "1st bike",
      bikeId: 'bike-1'
   }
]

// let's say we only have cars and bikes
itemValue = item => item.type === 'car' ? item.carId : item.bikeId
itemTitle = item => item.type === 'car' ? item.carTitle : item.bikeTitle
```

Reply:
> But you still need to allow a function to prop because

I've never disagreed with it. I've only said that it's possible to call `key` (if it's a function) directly without any proxy

Reply:
> it's possible to call key (if it's a function) directly without any proxy

Yes, but since in most cases `itemKey` is a string and not a function, a proxy function is used anyway. In method there is typeof check but in a computed prop I think there is a bigger overhead since it has to be cached and in order to cache vue must watch changes in dependencies. For me looks like a performance battle between `typeof` operator and `vue`'s computed properties, and I'll bet on `typeof`.

Anyway, I'll leave it as a method for now, if anyone wants it as a computed prop can also send a PR.